activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -533,8 +533,57 @@ def insert_fixtures(*)
         without_sql_mode("NO_AUTO_VALUE_ON_ZERO") { super }
       end
 
+      def insert_fixtures_set(fixture_set, tables_to_delete = [])
+        iterate_over_results = -> { while raw_connection.next_result; end; }
+
+        with_multi_statements do
+          without_sql_mode("NO_AUTO_VALUE_ON_ZERO") do
+            super(fixture_set, tables_to_delete, &iterate_over_results)
+          end
+        end
+      end
+
       private
 
+        def combine_multi_statements(fixture_inserts, tables_to_delete)
+          super
+
+          fixture_inserts.each_with_object([]) do |sql, total_sql|
+            previous_packet = total_sql.last
+            sql << ";\n"
+            if max_allowed_packet_reached?(sql, previous_packet) || total_sql.empty?
+              total_sql << sql
+            else
+              previous_packet << sql
+            end
+          end
+        end
+
+        def max_allowed_packet_reached?(current_packet, previous_packet)
+          if current_packet.bytesize > max_allowed_packet
+            raise ActiveRecordError, "Fixtures set is too large #{current_packet.bytesize}. Consider increasing the max_allowed_packet variable."
+          elsif previous_packet.nil?
+            false
+          else
+            (current_packet.bytesize + previous_packet.bytesize) > max_allowed_packet
+          end
+        end
+
+        def max_allowed_packet
+          @max_allowed_packet ||= execute("SELECT @@SESSION.max_allowed_packet").first[0].to_i
```

Comment:
How about using `@max_allowed_packet ||= show_variable("max_allowed_packet")`?

Reply:
very cool, wasn't aware of this method!